src/app/src/components/Layers/ReferenceImageLayer.js

```diff
@@ -36,8 +52,26 @@ export default function ReferenceImageLayer() {
         [images]
     );
 
+    const updateReferenceImageDebounced = useTrailingDebounceCallback({
+        callback: (id, boundary, layer) => {
+            postReferenceImage(
+                convertImageMetadataFromStateFormat(id, boundary, layer)
+            );
+        },
+        immediateCallback: (id, boundary, layer) => {
+            dispatch(
+                api.util.updateQueryData(
+                    'updateReferenceImage',
```

Comment:
I think this should be `getBoundaryDetails` instead of `updateReferenceImage`. When the reference image is updated, we're changing what would be returned from `getBoundaryDetails`, so this is the cache that should be updated.

If we can update the `getBoundaryDetails` cache like this in "real time" (i.e. without waiting to query the django server), then the cache will operate effectively as the store for boundary details. The map slice then need only store ephemeral state related to user interface (e.g. visibility of reference layers, whether the UI is in add polygon mode). 

I realize I'm requesting a fairly significant change, but I think having a single source of truth will save us a lot of headache later on. Let me know if you want talk about it. Also looping in @rajadain for his input.